engine/topology/src/test/java/org/opennms/oce/engine/topology/model/UpdateInventoryTest.java

```diff
@@ -26,159 +26,166 @@
-        final ModelObjectImpl port2 = new ModelObjectImpl("Port", "n33-c1-p2");
-        final ModelObjectImpl port3 = new ModelObjectImpl("Port", "n33-c1-p3");
-        final ModelObjectImpl port4 = new ModelObjectImpl("Port", "n33-c1-p4");
+        model = inventoryManager.getModel();
+        root = model.getRoot();
 
-        card.setParent(device);
+        assertThat(root.getChildren(), hasSize(0));
+        assertThat(root.getType(), is("MODEL"));
+    }
 
-        port1.setParent(card);
-        port2.setParent(card);
-        port3.setParent(card);
-        port4.setParent(card);
+    @Test
+    public void canLoadSimpleInventory() {
+        inventoryManager.clean();
 
-        model.printModel();
+        TopologyInventory inventory = new TopologyInventory();
+        InventoryObject obj = new InventoryObjectBean("DEVICE", "n1", null, "MODEL", "model");
+        inventory.addObject(obj);
+        inventoryManager.loadInventory(inventory);
 
-        model.addObject(device);
+        model = inventoryManager.getModel();
+        ModelObject root = model.getRoot();
 
-        model.printModel();
+        assertThat(root.getChildren(), hasSize(1));
+        assertThat(root.getType(), is("MODEL"));
+    }
 
-        model.removeObjectById(device.getType(), device.getId());
+    @Test
+    public void canLoadDefaultInventory() {
+        inventoryManager.clean();
+
+        TopologyInventory inventory = new TopologyInventory();
+        InventoryObject objDevice = new InventoryObjectBean("DEVICE", "n1", null, "MODEL", "model");
+        inventory.addObject(objDevice);
+        InventoryObject objCard = new InventoryObjectBean("CARD", "n1-c1", null, "DEVICE", "n1");
+        inventory.addObject(objCard);
+        InventoryObject objPort1 = new InventoryObjectBean("PORT", "n1-c1-p1", null, "CARD", "n1-c1");
+        inventory.addObject(objPort1);
+        InventoryObject objPort2 = new InventoryObjectBean("PORT", "n1-c1-p2", null, "CARD", "n1-c1");
+        inventory.addObject(objPort2);
+        inventoryManager.loadInventory(inventory);
+
+        model = inventoryManager.getModel();
+        ModelObject root = model.getRoot();
+
+        assertThat(root.getChildren(), hasSize(1));
+        assertThat(root.getChildren(), hasSize(1));
+        assertThat(root.getType(), is("MODEL"));
+        assertThat(model.getObjectById("DEVICE", "n1"), notNullValue());
+        assertThat(model.getObjectById("CARD", "n1-c1"), notNullValue());
+        assertThat(model.getObjectById("PORT", "n1-c1-p1"), notNullValue());
+        assertThat(model.getObjectById("PORT", "n1-c1-p2"), notNullValue());
+    }
 
-        model.printModel();
+    @Test
+    public void canLoadInventoryWithTopology_plus_Link() {
+        inventoryManager.clean();
+
+        TopologyInventory inventory = new TopologyInventory();
+
+        InventoryObject objDevice1 = new InventoryObjectBean("DEVICE", "n1", null, "MODEL", "model");
```

Comment:
The `MockInventoryBuilder` could also be used to define the inventory here with less verbosity.